dev-python/PyQt6/PyQt6-6.3.0.ebuild

```diff
@@ -0,0 +1,174 @@
+# Copyright 2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{8..10} )
+inherit flag-o-matic python-r1 qmake-utils toolchain-funcs
+
+QT_PV="6.3:6" # minimum tested qt version
+
+DESCRIPTION="Python bindings for the Qt framework"
+HOMEPAGE="https://www.riverbankcomputing.com/software/pyqt/"
+SRC_URI="mirror://pypi/${P::1}/${PN}/${P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64"
+# keep defaults in sync with qtbase, slow build so preferably not more
+IUSE="
+	+dbus debug qml designer examples gles2-only +gui help
+	multimedia +network opengl positioning printsupport quick quick3d
+	+sql +ssl svg testlib webchannel websockets +widgets +xml"
+# see `grep -r "%Import " sip` and `grep qmake_QT project.py`
+REQUIRED_USE="
+	${PYTHON_REQUIRED_USE}
+	designer? ( gui widgets )
+	help? ( gui widgets )
+	multimedia? ( gui network )
+	opengl? ( gui )
+	printsupport? ( gui widgets )
+	qml? ( network )
+	quick3d? ( gui qml )
+	quick? ( gui qml )
+	sql? ( widgets )
+	svg? ( gui )
+	testlib? ( gui widgets )
+	webchannel? ( network )
+	websockets? ( network )
+	widgets? ( gui )"
+
+DEPEND="
+	${PYTHON_DEPS}
+	>=dev-qt/qtbase-${QT_PV}[dbus?,gles2-only=,gui?,network?,opengl?,sql?,ssl=,widgets?,xml?]
+	dbus? (
+		dev-python/dbus-python[${PYTHON_USEDEP}]
+		sys-apps/dbus
+	)
+	designer? ( >=dev-qt/qttools-${QT_PV} )
+	help? ( >=dev-qt/qttools-${QT_PV} )
+	multimedia? ( >=dev-qt/qtmultimedia-${QT_PV} )
+	positioning? ( >=dev-qt/qtpositioning-${QT_PV} )
+	qml? ( >=dev-qt/qtdeclarative-${QT_PV} )
+	quick3d? ( >=dev-qt/qtquick3d-${QT_PV} )
+	svg? ( >=dev-qt/qtsvg-${QT_PV} )
+	webchannel? ( >=dev-qt/qtwebchannel-${QT_PV} )
+	websockets? ( >=dev-qt/qtwebsockets-${QT_PV} )"
+RDEPEND="
+	${DEPEND}
+	>=dev-python/PyQt6_sip-13.2[${PYTHON_USEDEP}]"
+BDEPEND="
+	${PYTHON_DEPS}
+	>=dev-python/PyQt-builder-1.11[${PYTHON_USEDEP}]
+	>=dev-python/sip-6.5[${PYTHON_USEDEP}]
+	>=dev-qt/qtbase-${QT_PV}
+	dbus? ( virtual/pkgconfig )"
+
+src_prepare() {
+	default
+
+	# qmake6 needs real g++ (not clang), but at least respect ${CHOST} #726112
+	mkdir "${T}"/cxx || die
```

Comment:
If genuinely does need g++ then BDEPEND on sys-devel/gcc is going to be important for upcoming Clang projects

Reply:
Haven't dug into it, but from what I seen It's parsing g++'s output, which fails with other compilers which are different -- and good point, guess qmake-everything will actually need to BDEPEND on gcc if that's how it is.

Reply:
Originally I had a wrapper with $(tc-getCXX), it was all fine and dandy until I tried clang ;p